forem/forem
# Allow Moderator Role to adjust post tags

## What type of PR is this? (check all applicable)


## Description
This PR extends the abilities of the Moderator role to adjusting the tags on a post.

## Related Tickets & Documents
- Closes https://github.com/forem/forem/issues/17443
- Closes https://github.com/forem/forem/issues/17705

## QA Instructions, Screenshots, Recordings
1. As an admin, enable the `moderator_role` feature flag (`/admin/feature_flags/features`)
2. Choose 2 users you have login info for(`/admin/member_manager_users`), and make them a Moderator and Trusted.
3. Visit a chosen tag page in `/admin` and add a third user to it as a Tag Moderator.
4. In an incognito window, sign in as said Moderator user.
5. Navigate to an article and open the Moderator Actions Panel. Click the Adjust Tags dropdown.
6. Type in a valid tag and click "submit". The added tag should display in the post's tag list.
7. Click the "delete" icon next to a tag, type in a reason and click "Submit". The tag should be removed from the post successfully.
8. In another incognito window, sign in as said Trusted user, navigate to post and open the Moderation Panel. You should NOT see the Adjust Tags dropdown.
9. In a third incognito window, sign in as said Tag Moderator and navigate to an article that has the chosen tag; you should see the Adjust tags section. For an article without chosen tag, you should NOT see the Adjust tags section.

### UI accessibility concerns?
Will be addressed as identified.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?
